### PR TITLE
Add lintrunner to test infra

### PR DESCRIPTION
**Pull request by PaliC**
Add lintrunner to test infra.

Outside of copying lintrunner we do the following things

- Lint lintrunner (lol)
- Add lintrunner's tests to test infra
- Change colors of warning/error messages (as I just figured out how to do that)
- Updated changelog to prep for release

TODO: We should also enable linting with all of the linters in the repo in the `Lint.yml` job. Right now we just use RUSTFMT and MYPY as we only checked those 2 consistently.

### CODE REVIEW

File: .github/workflows/lintrunner_ci.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
+name: Lintrunner CI
+
+on:
+  pull_request:
```
Comment (seemethere) [inline .github/workflows/lintrunner_ci.yml:4]: We might want to limit this CI to just when things in the `lintrunner` folder get edited

**Review by seemethere — CHANGES_REQUESTED** [review]
Can we put the `lintrunner` directory in the `tools` directory?

File: .github/workflows/lintrunner_ci.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    - name: Upload wheels
+      uses: actions/upload-artifact@v2
+      with:
+        name: wheels
+        path: dist
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    if: "startsWith(github.ref, 'refs/tags/')"
```
Comment (huydhn) [inline .github/workflows/lintrunner_ci.yml:111]: As we have both lintrunner and torchfix in the same repo, does this make sense to have more specific release tags for them?  `refs/tags/v*` works well if lintrunner stands alone.  A simple tweak would be to have something like `refs/tags/lintrunner/v*`, any thoughts? This doesn't need to be done in the same PR though IMO but we probably need it later

**Review by huydhn — APPROVED** [review]

**Comment by PaliC** [comment]
> Can we put the `lintrunner` directory in the `tools` directory?

moved

**Review by seemethere — APPROVED** [review]

**Pull request merged**